src/Gui/Selection/Selection.cpp

```diff
@@ -311,6 +311,101 @@ std::vector<SelectionSingleton::SelObj> SelectionSingleton::getPickedList(const
     return temp;
 }
 
+std::vector<Gui::SelectionObject> SelectionSingleton::getSelectionIn(App::DocumentObject* container,

```

Comment:
This seems like a long complicated function. Could logic simplified by having a helper function with early return?

Reply:
I'm not sure, I can't see obvious way to do this

Reply:
Remind me